### PR TITLE
Add support for MPL 2.0

### PR DESCRIPTION
**Pull request by CalebAlbers**
This PR adds support for the Mozilla Public License v2.0.

I took the liberty of refactoring the list of licenses in the README to use a bulleted list, but if you'd prefer me to stick to the existing one-line list, let me know and I'll fix that.

Thanks!

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,415 @@
 {
 	"name": "copyright-inserter",
-	"version": "1.0.12",
-	"lockfileVersion": 1,
+	"version": "1.0.13",
+	"lockfileVersion": 2,
```
Comment (minherz) [inline package-lock.json]: what is a reason for upgrading the locker version?

File: src/copyrightinserter.ts [hunk 2]
```diff
@@ -97,7 +98,14 @@ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU Affero General Public License for more details.
 
 You should have received a copy of the GNU Affero General Public License
-along with this program.  If not, see <https://www.gnu.org/licenses/>.`]
+along with this program.  If not, see <https://www.gnu.org/licenses/>.`],
+
+        ['mpl2', (holder: string, year:string) =>
+`Copyright (C) ${year} ${holder}
```
Comment (minherz) [inline src/copyrightinserter.ts:104]: MPL is copyleft license. It does not require and should not have `Copyright...` line.

**Review by minherz — CHANGES_REQUESTED** [review]
Please, remove `Copyright`, change documentation to emphasis that MPL is _copyleft_ and not _copyright_ license.
It is important to me to understand the reason for upgrading the version of the locker file as well.

File: package-lock.json (hunk 1 above)
Comment (CalebAlbers) [reply package-lock.json]: That happened automatically due to me using a newer version of NPM - I'll revert and fix that.

File: src/copyrightinserter.ts (hunk 2 above)
Comment (CalebAlbers) [reply src/copyrightinserter.ts:104]: Thanks for the quick review! Although MPL is a copyleft license, it does not preclude copyright ownership for source code. The [MPL 2.0 FAQ](https://www.mozilla.org/en-US/MPL/2.0/FAQ/#apply) page talks about this under Question 4:

> You may also add additional accurate notices of copyright ownership, such as the name of the copyright holder, but this is not necessary.

Comment (minherz) [reply src/copyrightinserter.ts:104]: It sounds like your use case utilizes [Q4](view-source:https://www.mozilla.org/en-US/MPL/2.0/FAQ/#apply) clarifications. 
My concern was that the current implementation creates controversy between inserting a copyleft comment and needing extra keystrokes to delete copyright that is not a part of the notice.
I am OK with leaving this notice. However, if there will be requirements to align the text with the standard (and not optional) form of the license, this line will be removed or made opted-in.

**Review by minherz — APPROVED** [review]
lgtm. thank you for contribution.

**Comment by minherz** [comment]
@CalebAlbers , I will release the next version of the plugin in the Market during this week.

File: src/copyrightinserter.ts (hunk 2 above)
Comment (CalebAlbers) [reply src/copyrightinserter.ts:104]: @minherz that sounds reasonable to me -  a future option to preclude a copyright line would also be applicable to the two GPL licenses as well. Care would have to be taken to rework the `hasCopyright` function though, as it currently uses regex to search for the word `Copyright` (not a blocker, just something to keep in mind).

Thanks for the discussion and merge! I appreciate it.

**Pull request merged**